pkg/sink/codec/canal/canal_json_decoder.go

```diff
@@ -263,6 +274,52 @@ func (b *batchDecoder) assembleHandleKeyOnlyRowChangedEvent(
 	return b.NextRowChangedEvent()
 }
 
+func setColumnInfos(
+	tableInfo *timodel.TableInfo,
+	rawColumns map[string]interface{},
+	mysqlType map[string]string,
+	pkNames map[string]struct{},
+) {
+	mockColumnID := int64(100)
+	for name := range rawColumns {
+		columnInfo := new(timodel.ColumnInfo)
+		columnInfo.ID = mockColumnID
+		columnInfo.Name = pmodel.NewCIStr(name)
+		if utils.IsBinaryMySQLType(mysqlType[name]) {
+			columnInfo.AddFlag(mysql.BinaryFlag)
+		}
+		if _, isPK := pkNames[name]; isPK {
+			columnInfo.AddFlag(mysql.PriKeyFlag)
+		}
+		tableInfo.Columns = append(tableInfo.Columns, columnInfo)
+		mockColumnID++
+	}
+}
+
+func setIndexes(
+	tableInfo *timodel.TableInfo,
+	pkNames map[string]struct{},
+) {
+	indexColumns := make([]*timodel.IndexColumn, 0, len(pkNames))
+	for idx, col := range tableInfo.Columns {
+		name := col.Name.O
+		if _, ok := pkNames[name]; ok {
+			indexColumns = append(indexColumns, &timodel.IndexColumn{
+				Name:   pmodel.NewCIStr(name),
+				Offset: idx,
+			})
+		}
+	}
+	indexInfo := &timodel.IndexInfo{
+		ID:      1,
```

Comment:
Please follow the previous behavior. https://github.com/pingcap/tiflow/blob/master/cdc/model/sink.go#L917-L920

Reply:
there is only primary key, and should 1, it's the previous behavior.